lib/postPrepareHook.js

```diff
@@ -29,118 +20,31 @@ module.exports = function(context) {
-                        return Q();
-                    }
-
-                    // If new files got added or deleted, reload the whole app instead of specific files only
-                    // e.g: index.html references a logo file 'img/logo.png'
-                    // deleting the 'img/logo.png' file will trigger a reload that will remove it from the rendered app
-                    // likewise, adding the 'img/logo.png' file will trigger it to be shown on the app
-                    lr.browserSync.reload();
-                    return Q();
-                }).fail(function(err) {
-                    var msg = ' - An error occurred: ' + err;
-                    logger.show(msg);
-                    lr.stopServer();
-                });
-            }
-        }],
-        watchOptions: {
-
-            // If user specified files/folders to ignore (via `cordova run android -- --livereload --ignore=build/**/*.*`), ignore those files
-            // ... Otherwise, don't ignore any files (That's the default). The function below achieves that goal.
-            ignored: ignoreOption || function(string) {
-                return false;
-            },
-
-            // Ignore the initial add events .
-            // Don't run prepare on the initial addition of files,
-            // Only do it on subsequent ones
-            ignoreInitial: true
-        },
-        tunnel: livereloadOptions.tunnel || false,
-        ghostMode: livereloadOptions.ghostMode || true
-    });
-
-    return lr.startServer().then(function(server_url) {
-        return promiseUtils.Q_chainmap(options.platforms, function(plat) {
-            var platformIndexUrl = url.resolve(server_url, path.join(multiPlatforms.getPlatformWWWFolder(plat), startPage));
-            helpers.ChangeStartPage(projectRoot, plat, platformIndexUrl);
-        });
-    }).then(function() {
-        return promiseUtils.Q_chainmap(options.platforms, function(plat) {
-            var platWWWFolder = multiPlatforms.getPlatformWWWFolder(plat);
-
-            var platformIndexLocal = path.join(projectRoot, platWWWFolder, startPage);
-            var cspRemover = new CSPRemover(platformIndexLocal);
-            return cspRemover.Remove().then(function() {
-                var atsRemover = new ATSRemover(projectRoot, plat);
-                return atsRemover.Remove();
+    //  Start LiveReload server
+    var projectRoot = context.opts.projectRoot;
+    var platforms = context.opts.platforms;
+
+    return lr.start(projectRoot, platforms, {
+        cb: function (event, file, lrHandle) {
+            // After a file changes, first run `cordova prepare`, then reload.
+            context.cordova.raw.prepare().then(function () {
+                var patcher = new lr.Patcher(projectRoot, platforms);
+                return patcher.removeCSP();
+            }).then(function () {
+                if (event === 'change') {
+                    return lrHandle.reloadFile(file);
```

Comment:
Would reloading the single changed file work in case of hrefs (lets say, in iframes) referencing this changed file?